docs/src/user-manual/components.md

```diff
@@ -1,30 +1,165 @@
+sensitive information is only shared with authorized users and/or applications.
+
+[Keycloak](https://www.keycloak.org/) is currently used for Enduro's IAM
+component. Keycloak is based on standard protocols and provides support for
+OpenID Connect, OAuth 2.0, and SAML.
+
+## User interface
+
+Enduro's web-based user interface is the graphical method by which operators can
+interact with Enduro to see and manage ingest and preservation workflows, view
+information about packages, and more. Key information from other components and
+applications (such as tasks executed by the [preservation
+engine](#preservation-engine) in a workflow) is also collected via API and shown
+in the Enduro Ingest user interface when relevant.
+
+Enduro's user interface is built using [Vue.js](https://vuejs.org/) and
+[Bootstrap](https://getbootstrap.com/)
+
+## Preservation engine
+
+This component manages the transformation of Processing Information Packages
+(PIPs) into Archival Information Packages (AIPs) in a preservation workflow
+during processing. The preservation engine is equipped with a number of tools
+and microservices to perform preservation tasks (such as characterization,
+metadata capture, and normalization, etc.), and communicates the outcome of
+these actions back to Enduro for display in the
+[user interface](#user-interface).
+
+Enduro Ingest currently supports two different preservation engines:
+
+* [Archivematica](https://archivematica.org) is an open-source digital
+  preservation system that is designed to maintain standards-based, long-term
+  access to collections of digital objects. It provides an integrated suite of
+  free and open-source tools that allows users to process digital objects and
+  generate Archival Information Packages (AIPs) in compliance with the Open
+  Archival Information System
+  ([OAIS](https://en.wikipedia.org/wiki/Open_Archival_Information_System))
+  functional model and other digital preservation standards and best practices,
+  define preservation policies, manage processing configurations, and more.
+  Archivematica includes a web-based dashboard for user interaction, as well as
+  a set of [API](https://en.wikipedia.org/wiki/API) endpoints that Enduro uses
+  to integrate with Archivematica and display preservation task outcomes in the
+  Enduro user interface.
+* [a3m](https://github.com/artefactual-labs/a3m) is a streamlined version of
+  Archivematica that is wholly focused on AIP creation. It does not have
+  external dependencies, integration with access systems, search capabilities,
+  or a graphical interface. It was designed to reduce the bulk of
+  Archivematica's extraneous functions for users operating at a large scale who
+  are more focused on throughput of digital objects for preservation.
+
+## Preservation storage service
+
+This component acts as an interface between the
+[preservation engine](#preservation-engine) and whatever storage has been
```

Comment:
"whatever storage" -> "storage devices"?

Reply:
Done, thanks for the suggestion